src/clients/FtpClient.cc

```diff
@@ -285,7 +285,7 @@ Ftp::Client::failed(err_type error, int xerrno, ErrorState *err)
         ftperr->ftp.reply = xstrdup(reply);
 
     if (!err) {
-        fwd->request->detailError(error, xerrno);
+        request->detailError(error, xerrno);
```

Comment:
If this and similar `s/fwd->request/request/` changes are not necessary, then let's not do them in this issue. They increase noise and feel out of scope.